File: src/main/java/com/coffee/shop/api/demo/product/domain/api/dto/ProductFilter.java
package com.coffee.shop.api.demo.product.domain.api.dto;

import java.math.BigDecimal;
import lombok.Data;

@Data
public class ProductFilter {

    private BigDecimal priceFrom;
    private BigDecimal priceTo;
    private String productName;
    private String categoryCode;
    private int pageNumber;
    private int pageSize;
}

File: src/main/kotlin/com/coffee/shop/api/demo/product/domain/api/dto/ProductFilter.kt
package com.coffee.shop.api.demo.product.domain.api.dto

import lombok.Data
import java.math.BigDecimal

@Data
class ProductFilter {
    var priceFrom: BigDecimal? = null
    var priceTo: BigDecimal? = null
    var productName: String? = null
    var categoryCode: String? = null
    var colorCode: String? = null
    var pageNumber = 0
    var pageSize = 0
}